bin/broadcast_prs.py

```diff
@@ -0,0 +1,152 @@
+# The API endpoint for creating pull requests
+GITHUB_PR_URL_TEMPL = "https://api.github.com/repos/nf-core/{pipeline}/pulls"
+
+
+class UpdateTemplate:
+    """Updates the template content of an nf-core pipeline in
+    its `TEMPLATE` branch.
+
+    Args: - pipeline: The pipeline name
+          - branch: The template branch name, default=`TEMPLATE`
+          - token: GitHub auth token
+    """
+
+    def __init__(self, pipeline, branch='TEMPLATE', token=""):
+        """Basic constructor
+        """
+        self.pipeline = pipeline
+        self.repo_url = "{base_url}/{pipeline}".format(
+                    base_url=GH_BASE_URL.format(token=token),
+                    pipeline=pipeline)
+        self.branch = branch
+        self.tmpdir = tempfile.mkdtemp()
+        self.templatedir = tempfile.mkdtemp()
+        self.repo = None
+    
+    def run(self):
+        """Execute the template update.
+        """
+        config = self._clone_repo()
+        assert self.repo
+        context = utils.create_context(config)
+        self._apply_changes(context)
+        self._commit_changes()
+        self._push_changes()
+
+    def _clone_repo(self):
+        """Clone the repo and switch to the configured branch.
+        """
+        self.repo = git.Repo.clone_from(self.repo_url, self.tmpdir)
+        config = utils.fetch_wf_config(wf_path=self.tmpdir)
+        self.repo.git.checkout("origin/{branch}".format(branch=self.branch),
+            b="{branch}".format(branch=self.branch))
+        return config
+
+    def _apply_changes(self, context):
+        """Apply the changes of the cookiecutter template
+        to the pipelines template branch.
+        """
+        cookiecutter(NF_CORE_TEMPLATE,
+                     no_input=True,
+                     extra_context=context,
+                     overwrite_if_exists=True,
+                     output_dir=self.templatedir)
+        # Clear the template branch content
+        for f in os.listdir(self.tmpdir):
+            if f == ".git": continue
+            try:
+                shutil.rmtree(os.path.join(self.tmpdir, f))
+            except:
+                os.remove(os.path.join(self.tmpdir, f))
+        # Move the new template content into the template branch
+        template_path = os.path.join(self.templatedir, self.pipeline)
+        for f in os.listdir(template_path):
+            shutil.move(
+                os.path.join(template_path, f), # src
+                os.path.join(self.tmpdir, f), # dest
+            )
+
+    def _commit_changes(self):
+        """Commits the changes of the new template to the current branch.
+        """
+        self.repo.git.add(A=True)
+        self.repo.index.commit("Update nf-core pipeline template.")
+
+    def _push_changes(self):
+        self.repo.git.push()
+        
+
+def create_pullrequest(pipeline, origin="dev", template="TEMPLATE", token="", user="nf-core"):
+    """Create a pull request to a base branch (default: dev),
```

Comment:
I like that it defaults to dev, but I think it is still a lot of pipelines that do not use a dev branch yet.

Reply:
same as above. Let it fail. The dev branch will be a requirement we need to document :)